src/mam/mod_mam_rdbms_arch_async.erl

```diff
@@ -0,0 +1,190 @@
+-module(mod_mam_rdbms_arch_async).
+
+-include("mongoose_logger.hrl").
+
+-define(PM_PER_MESSAGE_FLUSH_TIME, [mod_mam_rdbms_async_pool_writer, per_message_flush_time]).
+-define(PM_FLUSH_TIME, [mod_mam_rdbms_async_pool_writer, flush_time]).
+-define(MUC_PER_MESSAGE_FLUSH_TIME, [mod_mam_muc_rdbms_async_pool_writer, per_message_flush_time]).
+-define(MUC_FLUSH_TIME, [mod_mam_muc_rdbms_async_pool_writer, flush_time]).
+
+-behaviour(gen_mod).
+-export([start/2, stop/1, config_spec/0, supported_features/0]).
+
+-export([archive_pm_message/3, archive_muc_message/3]).
+-export([flush_pm/2, flush_muc/2]).
+
+-type writer_type() :: pm | muc.
+
+-ignore_xref([archive_pm_message/3, archive_muc_message/3]).
+
+-spec archive_pm_message(_Result, mongooseim:host_type(), mod_mam:archive_message_params()) ->
+          ok | {error, timeout}.
+archive_pm_message(_Result, HostType, Params = #{archive_id := ArcID}) ->
+    PoolName = mongoose_async_pools:pool_name(HostType, 'pm-mam'),
+    wpool:cast(PoolName, {task, Params}, {hash_worker, ArcID}).
+
+-spec archive_muc_message(_Result, mongooseim:host_type(), mod_mam:archive_message_params()) ->
+    ok.
+archive_muc_message(_Result, HostType, Params0 = #{archive_id := RoomID}) ->
+    Params = mod_mam_muc_rdbms_arch:extend_params_with_sender_id(HostType, Params0),
+    PoolName = mongoose_async_pools:pool_name(HostType, 'muc-mam'),
+    wpool:cast(PoolName, {task, Params}, {hash_worker, RoomID}).
+
+%%%===================================================================
+%%% gen_mod callbacks
+%%%===================================================================
+-spec start(mongooseim:host_type(), [{writer_type(), gen_mod:module_opts()}]) -> any().
+start(HostType, Opts) ->
+    ?LOG_INFO(#{opts => Opts, what => starting_mam_async_pool, host_type => HostType}),
+    [ start_pool(HostType, Mod) || Mod <- Opts ].
+
+-spec stop(mongooseim:host_type()) -> any().
+stop(HostType) ->
+    ?LOG_INFO(#{what => stopping_mam_async_pool, host_type => HostType}),
+    Opts = gen_mod:get_module_opts(HostType, ?MODULE),
+    [ stop_pool(HostType, Mod) || Mod <- Opts ].
+
+-spec config_spec() -> mongoose_config_spec:config_section().
+config_spec() ->
+    mongoose_async_pools:config_spec().
+
+-spec supported_features() -> [atom()].
+supported_features() ->
+    [dynamic_domains].
+
+%%%===================================================================
```

Comment:
still these `=========` :D

Reply:
I honestly find them a nice visual separation between sections in code, but ok, it's just colours in the editor.